jena-arq/src/main/java/org/apache/jena/riot/out/JsonLDWriter.java

```diff
@@ -79,26 +130,85 @@ public void write(OutputStream out, DatasetGraph dataset, PrefixMap prefixMap, S
         IO.flush(w) ;
     }
 
-    private void serialize(Writer writer, DatasetGraph dataset, PrefixMap prefixMap, String baseURI) {
-        final Map<String, Object> ctx = new LinkedHashMap<>() ;
-        addProperties(ctx, dataset.getDefaultGraph()) ;
-        addPrefixes(ctx, prefixMap) ;
+    private JSONLD_FORMAT getOutputFormat() {
+	  		if ((RDFFormat.JSONLD_COMPACT_PRETTY.equals(format)) || (RDFFormat.JSONLD_COMPACT_FLAT.equals(format))) return JSONLD_FORMAT.COMPACT;
+	  		if ((RDFFormat.JSONLD_EXPAND_PRETTY.equals(format)) || (RDFFormat.JSONLD_EXPAND_FLAT.equals(format))) return JSONLD_FORMAT.EXPAND;
+	  		if ((RDFFormat.JSONLD_FLATTEN_PRETTY.equals(format)) || (RDFFormat.JSONLD_FLATTEN_FLAT.equals(format))) return JSONLD_FORMAT.FLATTEN;
+	  		if ((RDFFormat.JSONLD_FRAME_PRETTY.equals(format)) || (RDFFormat.JSONLD_FRAME_FLAT.equals(format))) return JSONLD_FORMAT.FRAME;
+	  		throw new RuntimeException("Unexpected output format");
+    }
+    
+    private boolean isPretty() {
```

Comment:
This may be down to personal preference, but I would find this a bit clearer with a constant collection, i.e.
```
private static final Set<RDFFormat> PRETTY_FORMATS = ImmutableSet.of(JSONLD_COMPACT_PRETTY, JSONLD_FLATTEN_PRETTY, etc.);
...
private boolean isPretty() { return PRETTY_FORMATS.contains(format); }
```